### ISSUE TITLE
[tests] Investigate TestSuiteCommon.TestGetPartialObjectMisAligned rare crash

### ISSUE THREAD

**Issue by vadmeste** [minio/minio#4160]
**This crash is found while compiling this PR https://github.com/minio/minio/pull/4150, However, commits which triggered this crash disappeared now.**

## Expected Behavior
Successful go test.

## Current Behavior
Test crashed.

```
Running all coverage for minio: ok  	github.com/minio/minio	0.009s	coverage: 66.7% of statements
?   	github.com/minio/minio/browser	[no test files]
----------------------------------------------------------------------
PANIC: server_test.go:1821: TestSuiteCommon.TestGetPartialObjectMisAligned
... Panic: runtime error: slice bounds out of range (PC=0x4621BC)
/home/travis/.gimme/versions/go1.7.5.linux.amd64/src/runtime/panic.go:458
  in gopanic
/home/travis/.gimme/versions/go1.7.5.linux.amd64/src/runtime/panic.go:34
  in panicslice
server_test.go:1875
  in TestSuiteCommon.TestGetPartialObjectMisAligned
/home/travis/.gimme/versions/go1.7.5.linux.amd64/src/reflect/value.go:302
  in Value.Call
/home/travis/.gimme/versions/go1.7.5.linux.amd64/src/runtime/asm_amd64.s:2086
  in goexit
OOPS: 200 passed, 1 PANICKED
--- FAIL: Test (13.62s)
FAIL
coverage: 74.9% of statements
FAIL	github.com/minio/minio/cmd	82.608s
make: *** [coverage] Error 1
```

## Possible Solution
No idea

## Context
Bug found

## Your Environment
All

**Comment by deekoder** [comment]
@hackintoshrao - please take a look at this.

**Comment by hackintoshrao** [comment]
- It looks like the issue has originated from [this line](https://github.com/vadmeste/minio/blob/issues/4138/cmd/server_test.go#L1875) where the contents of the slice has been accessed. 
- The size of the string slice depends on the output of the random function set in the loop (and inside the loop too) [here](https://github.com/vadmeste/minio/blob/issues/4138/cmd/server_test.go#L1849). 
- In cases where the random number generators used returns small numbers (0,1), there is the possibility wherein the slice would have very few elements in it and this will result in out of bound error. 
- We might have to guarantee the presence of 13 bytes of characters atleast in the slice.

**Comment by harshavardhana** [comment]
> In cases where the random number generators used returns small numbers (0,1), there is the possibility wherein the slice would have very few elements in it and this will result in out of bound error.
> We might have to guarantee the presence of 13 bytes of characters atleast in the slice.

:+1:

**Comment by hackintoshrao** [comment]
This is fixed, can be closed.

**Comment by harshavardhana** [comment]
Fixed in #4175
